queries/typespec/highlights.scm

```diff
@@ -0,0 +1,125 @@
+(identifier_or_member_expression) @type
+
+[
+  "import"
+  "using"
+  "namespace"
+  "model"
+  "is"
+  "extends"
+  "valueof"
+  "scalar"
+  "interface"
+  "op"
+  "enum"
+  "alias"
+  "dec"
+  "fn"
+  "extern"
+  "union"
+] @keyword
```

Comment:
Some of these should be more specific (read CONTRIBUTING.md).

Reply:
Managed to find fitting categories for all keywords.

Reply:
Some does not mean all, but this seems fine.

Reply:
Yeah, didn't necessarily plan on moving all but they all fit pretty well into one of the categories :sweat_smile: